x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.tsx

```diff
@@ -0,0 +1,144 @@
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  CredentialsLogic,
+  ICredentialsLogicActions,
+  ICredentialsLogicValues,
+} from '../credentials_logic';
+import { IApiToken } from '../types';
+import { TOKEN_TYPE_DISPLAY_NAMES } from '../constants';
+import { apiTokenSort } from '../utils/api_token_sort';
+import { getModeDisplayText, getEnginesDisplayText } from '../utils';
+
+export const CredentialsList: React.FC = () => {
+  const { deleteApiKey, fetchCredentials, showCredentialsForm } = useActions(
+    CredentialsLogic
+  ) as ICredentialsLogicActions;
+
+  const { apiTokens, meta } = useValues(CredentialsLogic) as ICredentialsLogicValues;
+
+  const items = apiTokens.slice().sort(apiTokenSort);
+
+  const columns = [
+    {
+      name: 'Test',
+      width: '12%',
+      render: (item: IApiToken) => {
+        return <div>{item.name}</div>;
+      },
+    },
+    {
+      name: 'Type',
+      width: '18%',
+      render: (item: IApiToken) => {
+        return <div>{TOKEN_TYPE_DISPLAY_NAMES[item.type]}</div>;
+      },
+    },
+    {
+      name: 'Key',
+      width: '43%',
+      render: (item: IApiToken) => {
+        return (
+          <div>
+            <EuiCopy
+              textToCopy={item.key || ''}
+              afterMessage={i18n.translate('xpack.enterpriseSearch.appSearch.credentials.copied', {
+                defaultMessage: 'Copied',
+              })}
+            >
+              {(copy) => (
+                <div>
+                  <EuiButtonIcon
+                    onClick={copy}
+                    iconType="copyClipboard"
+                    aria-label={i18n.translate(
+                      'xpack.enterpriseSearch.appSearch.credentials.copyApiKey',
+                      {
+                        defaultMessage: 'Copy API Key to clipboard',
+                      }
+                    )}
+                  />
+                  {item.key}
+                </div>
+              )}
+            </EuiCopy>
+          </div>
+        );
+      },
+    },
+    {
+      name: 'Modes',
+      width: '10%',
+      render: (item: IApiToken) => {
+        return <div>{getModeDisplayText(item)}</div>;
+      },
+    },
+    {
+      name: 'Engines',
+      width: '10%',
+      render: (item: IApiToken) => {
+        return <div>{getEnginesDisplayText(item)}</div>;
+      },
+    },
+    {
+      width: '4%',
+      render: (item: IApiToken) => {
+        return (
+          <EuiButtonIcon
+            onClick={() => showCredentialsForm(item)}
+            iconType="pencil"
+            aria-label={i18n.translate('xpack.enterpriseSearch.appSearch.credentials.editKey', {
+              defaultMessage: 'Edit API Key',
+            })}
+          />
+        );
+      },
+    },
+    {
+      width: '4%',
+      render: (item: IApiToken) => {
+        return (
+          <EuiButtonIcon
+            onClick={() => item.name && deleteApiKey(item.name)}
```

Comment:
Just curious btw, do we need the `item.name &&` check?

Reply:
Probably not 🤔 . "name" is a required field of IApiToken. I think it's safe to leave it off.

Reply:
5d0ab10